services/api/service.go

```diff
@@ -493,6 +527,84 @@ func (api *RelayAPI) startValidatorRegistrationDBProcessor() {
 	}
 }
 
+// simulateBlock sends a request for a block simulation to blockSimRateLimiter.
+func (api *RelayAPI) simulateBlock(ctx context.Context, opts blockSimOptions) (error, error) {
+	t := time.Now()
+	reqErr, simErr := api.blockSimRateLimiter.Send(ctx, opts.req, opts.isHighPrio, opts.fastTrack)
+	log := opts.log.WithFields(logrus.Fields{
+		"duration":   time.Since(t).Seconds(),
+		"numWaiting": api.blockSimRateLimiter.CurrentCounter(),
+	})
+	if simErr != nil {
+		ignoreError := simErr.Error() == ErrBlockAlreadyKnown || simErr.Error() == ErrBlockRequiresReorg || strings.Contains(simErr.Error(), ErrMissingTrieNode)
+		if !ignoreError {
+			// Mark builder as non-optimistic.
+			opts.builder.status.IsOptimistic = false
+			log.WithError(simErr).Error("block validation failed")
+			return nil, simErr
+		}
+	}
+	if reqErr != nil {
+		log.WithError(reqErr).Error("block validation failed: request error")
```

Comment:
should log as Warn or Info. Error means something needs to be investigated by the devops team! (i.e. only system errors should log with error level, never user errors)

Reply:
changed to warn!